modules/llrt_http/src/incoming.rs

```diff
@@ -0,0 +1,354 @@
+
+        test_async_with_opts(
+            |ctx| {
+                crate::init(&ctx).unwrap();
+                Box::pin(async move {
+                    let globals = ctx.globals();
+                    let run = async {
+                        let fetch: Function = globals.get("fetch")?;
+
+                        let options = Object::new(ctx.clone())?;
+                        options.set("method", "GET")?;
+
+                        let url = format!("http://{}/some-path/", mock_server.address().clone());
+
+                        // The scope ensure we drop all responses
+                        {
+                            let response_promise: Promise = fetch.call((url, options.clone()))?;
+                            let response: Class<Response> = response_promise.into_future().await?;
+                            let mut response = response.borrow_mut();
+                            let _response2 = response.clone(ctx.clone()).unwrap();
+                        }
+
+                        tokio::task::yield_now().await;
+
+                        Ok(())
+                    };
+                    run.await.catch(&ctx).unwrap();
+                })
+            },
+            TestOptions::new().no_pending_jobs(),
+        )
+        .await;
+    }
+
+    #[tokio::test]
+    async fn test_incoming_lagged() {
+        let mock_server = MockServer::start().await;
+        let welcome_message = vec![b'x'; 1024 * 1024 * 50];
+
+        Mock::given(matchers::path("some-path/"))
+            .respond_with(ResponseTemplate::new(200).set_body_bytes(welcome_message.clone()))
+            .mount(&mock_server)
+            .await;
+
+        test_async_with_opts(|ctx| {
+            crate::init(&ctx).unwrap();
+            Box::pin(async move {
+                let globals = ctx.globals();
+                let run = async {
+                    let fetch: Function = globals.get("fetch")?;
+
+                    let options = Object::new(ctx.clone())?;
+                    options.set("method", "GET")?;
+
+                    let url = format!("http://{}/some-path/", mock_server.address().clone());
+
+                    let response_promise: Promise = fetch.call((url, options.clone()))?;
+                    let response: Class<Response> = response_promise.into_future().await?;
+                    let mut response = response.borrow_mut();
+                    let mut response2 = response.clone(ctx.clone()).unwrap();
+
+                    let response_text = response.text(ctx.clone()).await.unwrap();
+                    assert_eq!(response.status(), 200);
+                    assert_eq!(response_text.as_bytes(), welcome_message);
+
+                    let response2_err = response2.text(ctx.clone()).await.catch(&ctx).unwrap_err();
+                    assert_eq!(response2.status(), 200);
+                    assert!(matches!(response2_err, CaughtError::Exception(e) if e.message().unwrap() == ERROR_MSG_BROADCAST_LAGGED));
```

Comment:
Why do we lagg behind here? This would be possible no? 
```javascript
const a = await fetch("https://www.amazon.com")
const b = a.clone();
console.log(await a.text(), await b.text())
```

Reply:
So lag is due to our implementation of broadcast. Broadcast uses a bounded channel, I set it to 16. Your example should work, but it is not always guaranteed to, it depends on how many frames are in the fetch.

The official implementation mentions that it would buffer indefinitely but also says that it is not ideal for readers to lag behind.

This implementation is a middle ground, I expect we will rip it off once we have proper streams.